sw/host/opentitanlib/src/io/jtag.rs

```diff
@@ -0,0 +1,88 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::Result;
+use serde::{Deserialize, Serialize};
+use std::rc::Rc;
+use structopt::StructOpt;
+use thiserror::Error;
+
+use crate::app::TransportWrapper;
+use crate::dif::lc_ctrl::{DifLcCtrlState, LcCtrlReg};
+use crate::impl_serializable_error;
+
+#[derive(Debug, StructOpt)]
+pub struct JtagParams {
+    #[structopt(long, help = "OpenOCD binary path.", default_value = "openocd")]
+    pub openocd: String,
+
+    #[structopt(long, help = "Path to OpenOCD JTAG adapter config file.")]
+    openocd_adapter_config: String,
+}
+
+impl JtagParams {
+    pub fn create(&self, transport: &TransportWrapper) -> Result<Rc<dyn Jtag>> {
+        let jtag = transport.jtag(&self.openocd, &self.openocd_adapter_config)?;
+        Ok(jtag)
+    }
+}
+
+/// Errors related to the JTAG interface.
+#[derive(Error, Debug, Deserialize, Serialize)]
+pub enum JtagError {
+    #[error("JTAG timeout")]
+    Timeout,
+    #[error("JTAG busy")]
+    Busy,
+    #[error("Generic error {0}")]
+    Generic(String),
+}
+impl_serializable_error!(JtagError);
+
+/// Represents an OpenOCD command sent over JTAG.
+pub enum OpenOcdCmd {
+    ReadLcCtrlReg(LcCtrlReg),
+    WriteLcCtrlReg(LcCtrlReg, u32),
+    // ReadCpuReg(IbexReg),
+    // WriteCpuReg(IbexReg, u32),
+    ReadMemory { addr: u32 },
+    WriteMemory { addr: u32, value: u32 },
+    Halt,
+    Resume,
+    Shutdown,
+    Version,
```

Comment:
Should we add the `reset halt` command?

Reply:
We will re-assess the exact set of commands as we implement the manufacturing tests. It might make sense to have a general `Reset` command that takes a parameter but we need to think about its mixed usage for the LC tap and the RISC-V tap.